mir-module/src/main/resources/xsl/mods2oaiheader.xsl

```diff
@@ -28,6 +37,49 @@
     <xsl:value-of select="substring-before(service/servdates/servdate[@type='modifydate'],'T')" />
   </datestamp>
 
+  <!-- check for the following sets: doc-type,open_access,openaire,driver,ec_fundedresources,GENRE,ddc -->
+  <setSpec><xsl:value-of select="concat('doc-type:', substring-after(metadata/def.modsContainer/modsContainer/mods:mods/mods:classification[@generator='mir_genres2diniPublType-mycore']/@valueURI,'#'))" /></setSpec>
+  <setSpec><xsl:value-of select="concat('GENRE:', substring-after(metadata/def.modsContainer/modsContainer/mods:mods/mods:genre/@valueURI,'#'))" /></setSpec>
+
+  <xsl:choose>
+    <xsl:when test="metadata/def.modsContainer/modsContainer/mods:mods/mods:classification[@authority='ddc']">
+      <setSpec><xsl:value-of select="concat('ddc:', metadata/def.modsContainer/modsContainer/mods:mods/mods:classification[@authority='ddc'])" /></setSpec>
+    </xsl:when>
+    <xsl:when test="metadata/def.modsContainer/modsContainer/mods:mods/mods:classification[@authority='sdnb']">
+      <setSpec><xsl:value-of select="concat('ddc:', metadata/def.modsContainer/modsContainer/mods:mods/mods:classification[@authority='sdnb'])" /></setSpec>
+    </xsl:when>
+  </xsl:choose>
+
+  <xsl:variable name="isOpenAccess" xmlns:encoder="xalan://java.net.URLEncoder"
+                select="document(concat('solr:q=',encoder:encode(concat($MCR.OAIDataProvider.OAI2.MapSetToQuery.open_access, ' AND id:', @ID)), '&amp;rows=0'))/response/result/@numFound" />
+  <xsl:if test="$isOpenAccess &gt; 0">
+    <setSpec>open_access</setSpec>
+  </xsl:if>
+
+  <xsl:variable name="isDriver" xmlns:encoder="xalan://java.net.URLEncoder"
+                select="document(concat('solr:q=',encoder:encode(concat($MCR.OAIDataProvider.OAI2.MapSetToQuery.driver, ' AND id:', @ID)), '&amp;rows=0'))/response/result/@numFound" />
+  <xsl:if test="$isDriver &gt; 0">
+    <setSpec>driver</setSpec>
+  </xsl:if>
+
+  <xsl:variable name="isOpenAire" xmlns:encoder="xalan://java.net.URLEncoder"
+                select="document(concat('solr:q=',encoder:encode(concat($MCR.OAIDataProvider.OAI2.MapSetToQuery.openaire, ' AND id:', @ID)), '&amp;rows=0'))/response/result/@numFound" />
+  <xsl:if test="$isOpenAire &gt; 0">
+    <setSpec>openaire</setSpec>
+  </xsl:if>
+
+  <xsl:variable name="isEC_FundedResources" xmlns:encoder="xalan://java.net.URLEncoder"
```

Comment:
SOLR-based sets should be handled by a common template and different parameters: setSpec, query